target/linux/bmips/image/bcm63268.mk

```diff
@@ -100,3 +100,17 @@ define Device/sercomm_shg2500
   SERCOMM_SWVER := 3207
 endef
 TARGET_DEVICES += sercomm_shg2500
+
+define Device/smartrg_sr505n
+  $(Device/bcm63xx-cfe)
```

Comment:
I'm having difficulty extracting the cferam. When I run the stock firmware there is no cferam.000 in the root directory, and binwalk doesn't find a jffs2 filesystem in the flash dump. 

There is a block of lzma data at 0x36E4 that looks similar to other cferam.000 files, but it's missing the first 12 bytes of the header:

    00000000: 1000 001f 0000 0000 1000 00ac 0000 0000  ................
    00000010: 4346 4531 4346 4531 8063 4af8 8063 4b00  CFE1CFE1.cJ..cK.
    00000020: 8063 93c0 8064 a8a0 8060 1000 8063 93c0  .c...d...`...c..
    00000030: 8064 1370 0000 0000 0000 0000 8060 12bc  .d.p.........`..

The bytes before the compressed data look like it might have one of the addresses:

    000036d0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
    000036e0: 8060 1000 6d00 0040 00c0 8303 0000 0000  .`..m..@........
    000036f0: 0000 0800 2c3e 00de 9614 46b3 4dab 9893  ....,>....F.M...
    00003700: 910b 1bc9 20db 3d12 557e 7f0f 1bd4 b2df  .... .=.U~......

Is this a different kind of firmware image? Or am I looking in the wrong place? Any insight would be greatly appreciated. 


For reference, this is the full binwalk:

    DECIMAL       HEXADECIMAL     DESCRIPTION
    --------------------------------------------------------------------------------
    14052         0x36E4          LZMA compressed data, properties: 0x6D, dictionary size: 4194304 bytes, uncompressed size: 230336 bytes
    65536         0x10000         Broadcom 96345 firmware header, header size: 256, firmware version: "68", board id: "63168MBV_17AZZ", ~CRC32 header checksum: 0x6141E047, ~CRC32 data checksum: 0x2E0F7A4D
    65792         0x10100         Squashfs filesystem, little endian, non-standard signature, version 4.0, compression:gzip, size: 6316561 bytes, 1044 inodes, blocksize: 65536 bytes, created: 2015-05-18 21:03:34
    6385932       0x61710C        LZMA compressed data, properties: 0x6D, dictionary size: 4194304 bytes, uncompressed size: 3552512 bytes
    8388608       0x800000        Broadcom 96345 firmware header, header size: 256, firmware version: "68", board id: "63168MBV_17AZZ", ~CRC32 header checksum: 0x4CB2EF72, ~CRC32 data checksum: 0x9260BE47
    8388864       0x800100        Squashfs filesystem, little endian, non-standard signature, version 4.0, compression:gzip, size: 6426126 bytes, 1049 inodes, blocksize: 65536 bytes, created: 2015-12-22 20:52:16
    14815500      0xE2110C        LZMA compressed data, properties: 0x6D, dictionary size: 4194304 bytes, uncompressed size: 3552512 bytes
    
    36E4
    
    DECIMAL       HEXADECIMAL     DESCRIPTION
    --------------------------------------------------------------------------------
    16            0x10            CFE boot loader
    206052        0x324E4         Copyright string: "Copyright (C) 2000-2011 Broadcom Corporation."
    207692        0x32B4C         HTML document header
    207827        0x32BD3         HTML document footer
    213128        0x34088         CRC32 polynomial table, big endian
    225836        0x3722C         HTML document header
    227359        0x3781F         HTML document footer
    227372        0x3782C         HTML document header
    229026        0x37EA2         HTML document footer
    229040        0x37EB0         HTML document header
    229764        0x38184         HTML document footer
    
    61710C
    
    DECIMAL       HEXADECIMAL     DESCRIPTION
    --------------------------------------------------------------------------------
    1644744       0x1918C8        Certificate in DER format (x509 v3), header length: 4, sequence length: 4
    2736144       0x29C010        Linux kernel version 2.6.30
    2790448       0x2A9430        DES SP2, big endian
    2790960       0x2A9630        DES SP1, big endian
    2816320       0x2AF940        CRC32 polynomial table, little endian
    2826284       0x2B202C        CRC32 polynomial table, big endian
    3143320       0x2FF698        Neighborly text, "NeighborSolicitsipv6 ah init: can't add xfrm type"
    3143340       0x2FF6AC        Neighborly text, "NeighborAdvertisementsadd xfrm type"
    3145855       0x30007F        Neighborly text, "neighbor %.2x%.2x.%.2x:%.2x:%.2x:%.2x:%.2x:%.2x lost on port %d(%s)(%s)"
    
    E2110C
    
    DECIMAL       HEXADECIMAL     DESCRIPTION
    --------------------------------------------------------------------------------
    1644872       0x191948        Certificate in DER format (x509 v3), header length: 4, sequence length: 4
    2736144       0x29C010        Linux kernel version 2.6.30
    2790448       0x2A9430        DES SP2, big endian
    2790960       0x2A9630        DES SP1, big endian
    2816320       0x2AF940        CRC32 polynomial table, little endian
    2826284       0x2B202C        CRC32 polynomial table, big endian
    3143324       0x2FF69C        Neighborly text, "NeighborSolicitsipv6 ah init: can't add xfrm type"
    3143344       0x2FF6B0        Neighborly text, "NeighborAdvertisementsadd xfrm type"
    3145859       0x300083        Neighborly text, "neighbor %.2x%.2x.%.2x:%.2x:%.2x:%.2x:%.2x:%.2x lost on port %d(%s)(%s)"

Reply:
> I'm having difficulty extracting the cferam. When I run the stock firmware there is no cferam.000 in the root directory, and binwalk doesn't find a jffs2 filesystem in the flash dump.

@kylehendrydev does the flash dump contain OOB data?
If it does maybe that's the reason why binwalk doesn't find it.
How did you get/create that flash dump?
Can you send the flash dump to me via email or Github upload?

If you boot OpenWrt by TFTP you should be able to mount the other image as Read Only and get cferam from there too. Then you can just send it via scp to any other host.

Reply:
I extracted the flash with an IC clip. The part is NOR flash so there is no OOB data. 

Dump is attached
[SR505n-flashdump.gz](https://github.com/openwrt/openwrt/files/14550965/SR505n-flashdump.gz)

Reply:
> I extracted the flash with an IC clip. The part is NOR flash so there is no OOB data.

Excuse me, but then I was mistaken, there's no cferam file since the device doesn't have a NAND.
Sorry for the misunderstanding, my bad.